src/main/java/io/odpf/firehose/sink/prometheus/builder/TimeSeriesBuilder.java

```diff
@@ -0,0 +1,182 @@
+package io.odpf.firehose.sink.prometheus.builder;
+
+import io.odpf.firehose.config.PromSinkConfig;
+import io.odpf.firehose.exception.EglcConfigurationException;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import com.google.protobuf.DynamicMessage;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.Duration;
+
+import cortexpb.Cortex;
+
+import java.util.Properties;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static io.odpf.firehose.sink.prometheus.PromSinkConstants.*;
+
+/**
+ * Builder for Cortex TimeSeries.
+ *
+ * The buildTimeSeries(Message message, int partition) method should be synchronised if called from multiple threads
+ */
+public class TimeSeriesBuilder {
+
+    private Cortex.TimeSeries.Builder cortexTimeSeriesBuilder = Cortex.TimeSeries.newBuilder();
+    private Cortex.LabelPair.Builder cortexLabelBuilder = Cortex.LabelPair.newBuilder();
+    private Cortex.Sample.Builder cortexSampleBuilder = Cortex.Sample.newBuilder();
+
+    private Properties metricNameProtoIndexMapping;
+    private Properties labelNameProtoIndexMapping;
+    private Integer timestampIndex;
+    private Boolean isEventTimestampEnabled;
+
+    /**
+     * Instantiates a new cortex time series builder.
+     *
+     * @param config    the prometheus sink config
+     */
+    public TimeSeriesBuilder(PromSinkConfig config) {
+        metricNameProtoIndexMapping = config.getSinkPromMetricNameProtoIndexMapping();
+        labelNameProtoIndexMapping = config.getSinkPromLabelNameProtoIndexMapping();
+        timestampIndex = config.getSinkPromProtoEventTimestampIndex();
+        isEventTimestampEnabled = config.isEventTimestampEnabled();
+    }
+
+    /**
+     * build list of sorted cortex time series object.
+     *
+     * @param message                           the protobuf message
+     * @param partition                         the kafka partition where the message is consumed
+     * @return                                  list of sorted cortex time series object
+     * @throws InvalidProtocolBufferException   the exception on invalid protobuf
+     */
+    public List<Cortex.TimeSeries> buildTimeSeries(Message message, int partition) throws InvalidProtocolBufferException {
+        Map<String, Object> labelPair = getLabelMessage(message, labelNameProtoIndexMapping, partition);
+        List<Map<String, Object>> metricList = getMetricMessage(message, metricNameProtoIndexMapping);
+        List<Cortex.TimeSeries> timeSeriesList = new ArrayList<>();
+        Long metricTimestamp = getMetricTimestamp(message);
+        for (Map<String, Object> metricName : metricList) {
+            buildMetric((String) metricName.get(METRIC_NAME));
```

Comment:
here also we can do metricName.get(METRIC_NAME).toString(), instead of casting it to string.